src/App.tsx

```diff
@@ -1,18 +1,64 @@
 import React from 'react';
 import './App.css';
+import { useSearchParams } from 'react-router-dom';
+import { Items } from './components/Items';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams({});
```

Comment:
mb would better to set initial values here like
`useSearchParams({ page: '1', perPage: '5' })`